openmdao/components/add_subtract_comp.py

```diff
@@ -0,0 +1,235 @@
+    Use scaling factor -1 for subtraction.
+
+    Attributes
+    ----------
+    _add_systems : list
+    List of equation systems to be initialized with the system.
+    """
+
+    def __init__(self, output_name=None, input_names=None, vec_size=1, length=1,
+                 val=1.0, scaling_factors=None, **kwargs):
+        """
+        Allow user to create an addition/subtracton system with one-liner.
+
+        Parameters
+        ----------
+        output_name : str
+            (required) name of the result variable in this component's namespace.
+        input_names : iterable of str
+            (required) names of the input variables for this system
+        vec_size : int
+            Length of the first dimension of the input and output vectors
+            (i.e number of rows, or vector length for a 1D vector)
+            Default is 1
+        length : int
+            Length of the second dimension of the input and ouptut vectors (i.e. number of columns)
+            Default is 1 which results in input/output vectors of size (vec_size,)
+        scaling_factors : iterable of numeric
+            Scaling factors to apply to each input.
+            Use [1,1,...] for addition, [1,-1,...] for subtraction
+            Must be same length as input_names
+            Default is None which results in a scaling factor of 1 on
+            each input (element-wise addition)
+        val : float or list or tuple or ndarray
+            The initial value of the variable being added in user-defined units. Default is 1.0.
+        **kwargs : str
+            Any other arguments to pass to the addition system
+            (same as add_output method for ExplicitComponent)
+            Examples include units (str or None), desc (str)
+        """
+        super(AddSubtractComp, self).__init__()
+
+        self._add_systems = []
+
+        if isinstance(output_name, string_types):
+            self._add_systems.append((output_name, input_names, vec_size, length, val,
+                                      scaling_factors, kwargs))
+        elif isinstance(output_name, collections.Iterable):
+            raise NotImplementedError('Declaring multiple addition systems '
+                                      'on initiation is not implemented.'
+                                      'Use a string to name a single addition relationship or use '
+                                      'multiple add_output calls')
+        elif output_name is None:
+            pass
+        else:
+            raise ValueError(
+                "first argument to adder init must be either of type "
+                "`str' or 'None'")
+
+    def initialize(self):
+        """
+        Declare options.
+
+        Parameters
+        ----------
+        complex : Boolean
+            Set True to enable complex math (e.g. for complex step verification)
+        """
+        self.options.declare('complex', default=False,
```

Comment:
I think you can get rid of this option, and instead, in `compute` you can query if `outputs._alloc_complex` is True. That way you only allocate complex when you are in a complex step situation. I am away from my computer right now, so will verify this later, but I am pretty sure it works. We should probably document this so that users can use it in their comps (and maybe remove the underscore since it is being used public.)